lxd/db/instances.go

```diff
@@ -872,6 +872,49 @@ func (c *ClusterTx) GetInstanceSnapshotsWithName(ctx context.Context, project st
 	return instances, nil
 }
 
+// GetInstancesWithConfig returns all available instances with the given config key and value.
+func (c *ClusterTx) GetInstancesWithConfig(key string, value string) ([]cluster.Instance, error) {
```

Comment:
To further optimize this, we'd likely want a `GetLocalInstancesWithLocalConfig`.
That would let us restrict the nodes.id and would also clarify that we will not be expanding profiles.

Reply:
@masnax is such a query our best bet here or should we expand InstanceFilter to allow for filtering on config?

If we can do this efficiently through a filter, it would also let us easily filter our based on Node and Type too.

Reply:
Doing any of this through the generator would be less efficient than the one big join we have here. The generator basically does all its heavy lifting in `go`.

The current generator equivalent of this `GetInstancesWithConfig` function would be:
```go
instances, err := cluster.GetInstances(ctx, tx, cluster.InstanceFilter{})

filteredInstances := []cluster.Instance
for _, instance := range instances {
  config, err := cluster.GetInstanceConfig(ctx, tx, instance.ID)
  val, ok := config[key]
    if ok && val == value {
      filteredInstances = append(filteredInstances, instance)
    }
}

return filteredInstances
```

@tomponline for your instance ID suggestion, we could just make the `InstanceFilter` variadic so we can supply many of them. Then have the generator loop on them, appending an `AND` each round.